src/Types.hs

```diff
@@ -285,6 +301,27 @@ areUnifiable FuncTy{} _ = False
 areUnifiable a b | a == b    = True
           | otherwise = False
 
+-- Checks whether or not the kindedness of types match
+-- Kinds are polymorphic constructors such as (f a)
+-- Note that this disagrees with the notion of unifiablitity in areUnifiable
+checkKinds :: Ty -> Ty -> Bool
+-- Base < Higher
+checkKinds (FuncTy argTysA retTyA _) (FuncTy argTysB retTyB _) =
+  let argKinds = zipWith (<=) (map tyToKind argTysA) (map tyToKind argTysB)
+      retKinds = tyToKind retTyA <= tyToKind retTyB
+  in all (== True) (retKinds : argKinds)
+--checkKinds (StructTy _ argTysA) (StructTy _ argTysB) =
```

Comment:
All the commented code can be deleted--what we have is sufficient for checking interfaces, I'm not sure that any other checks are needed.

Reply:
Done 56d3b02